python/example_code/bedrock-agent/knowledge_bases/knowledge_base.py

```diff
@@ -0,0 +1,349 @@
+            err.response["Error"]["Code"],
+            err.response["Error"]["Message"],
+        )
+        raise
+# snippet-end:[python.example_code.bedrock-agent.update_knowledge_base]
+
+
+# snippet-start:[python.example_code.bedrock-agent.delete_knowledge_base]
+def delete_knowledge_base(bedrock_agent_client, knowledge_base_id):
+    """
+    Deletes a knowledge base.
+
+    Args:
+        bedrock_agent_client: The Boto3 Bedrock Agent client.
+        knowledge_base_id (str): The ID of the knowledge base to delete.
+
+    Returns:
+        bool: True if the deletion was successful.
+    """
+    try:
+        bedrock_agent_client.delete_knowledge_base(
+            knowledgeBaseId=knowledge_base_id
+        )
+        
+        logger.info("Deleted knowledge base: %s", knowledge_base_id)
+        return True
+    except ClientError as err:
+        logger.error(
+            "Couldn't delete knowledge base %s. Here's why: %s: %s",
+            knowledge_base_id,
+            err.response["Error"]["Code"],
+            err.response["Error"]["Message"],
+        )
+        raise
+# snippet-end:[python.example_code.bedrock-agent.delete_knowledge_base]
+
+
+# snippet-start:[python.example_code.bedrock-agent.list_knowledge_bases]
+def list_knowledge_bases(bedrock_agent_client, max_results=None):
+    """
+    Lists the knowledge bases in your AWS account.
+
+    Args:
+        bedrock_agent_client: The Boto3 Bedrock Agent client.
+        max_results (int, optional): The maximum number of knowledge bases to return.
+
+    Returns:
+        list: A list of knowledge base details.
+    """
+    try:
+        kwargs = {}
+        if max_results is not None:
+            kwargs["maxResults"] = max_results
+
+        # Initialize an empty list to store all knowledge bases
+        all_knowledge_bases = []
+        
+        # Use paginator to handle pagination automatically
+        paginator = bedrock_agent_client.get_paginator('list_knowledge_bases')
+        page_iterator = paginator.paginate(**kwargs)
+        
+        # Iterate through each page of results
+        for page in page_iterator:
+            all_knowledge_bases.extend(page.get('knowledgeBaseSummaries', []))
+            
+        logger.info("Found %s knowledge bases.", len(all_knowledge_bases))
+        return all_knowledge_bases
+    except ClientError as err:
+        logger.error(
+            "Couldn't list knowledge bases. Here's why: %s: %s",
+            err.response["Error"]["Code"],
+            err.response["Error"]["Message"],
+        )
+        raise
+# snippet-end:[python.example_code.bedrock-agent.list_knowledge_bases]
+
+
+def main():
+    """
+    Shows how to use the Bedrock Agent API to work with knowledge bases.
+    """
+    logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+
+    parser = argparse.ArgumentParser(
+        description="Work with knowledge bases in your AWS account."
+    )
+    parser.add_argument(
```

Comment:
Is it possible to run these steps together as a set? For example: Create a KnowledgeBase, List, Get, Update, then Delete? With the role created as part of the process? Otherwise the user has to know what the required Role will look like, and provide the arn. We try to have scenarios be self-contained. For more information, see the example [standards](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/general-code-example-standards). If this is not possible for some reason and an exception needs to be made, let us know.

Reply:
Hey @gausekha, did you have a chance to review Rachel's note here?

Reply:
@rlhagerm and @beqqrry-aws I just added the IAM role and a scenario: https://github.com/gausekha/aws-doc-sdk-examples/commit/0dec32e68876db07d125f47923c52d0a5721f69c. I am trying to run this locally but keep running into an error about specifying the region. Could you please take a look?

Reply:
Ah, I see - got it that makes sense. Shall update it, thanks for catching it! And yes, shall try to update my profile as well and get a clean build before pushing the next commit.

Reply:
@rlhagerm Apologies for the back and forth! I tried updating the ARNs. Here's the commit: https://github.com/awsdocs/aws-doc-sdk-examples/pull/7451/commits/9356a35f188e9b83a90edc1a27b0dabd99714408. Please LMK if any additional updates are needed. Thanks!

Reply:
@rlhagerm Please LMK once you have a chance to look at this. Thanks!

Reply:
Is the CollectionArn something the user is supposed to provide? It looks like what is in there now is a placeholder, which may be fine, but we should include some information on how to provide the correct value and/or replace it in the code.

Reply:
@rlhagerm Apologies again for the late reply! Yes you are right and it is a placeholder. I think we can leave it for now but yes a user needs to create an Opensearch collection and then create a KB by specifying the collection ARN. In the console, an user can create a KB using the quick-create option where the collection is automatically created for them but not when they use the API.

Reply:
@gausekha , Ok, just add a comment with those instructions and I'll approve it.

Reply:
Great, thanks! Could you please check the latest commit: https://github.com/awsdocs/aws-doc-sdk-examples/pull/7451/commits/4fcf37011b8beece9e159dd5df8766fe1c673a7f? Thanks!

Reply:
@gausekha yes, approved, thank you!
We will take it from here, all merging will happen on our side as soon as oncall gets to it. Thank you for your patience.

Reply:
Great, thank you so much! Is it possible for you to share with me how long it takes generally for the changes to be merged, @rlhagerm ?